### PR TITLE
Add test to sparse weights module

### PR DESCRIPTION
**Pull request by lscheinkman**
@subutai Added tests to SparseWeights. Please review

### CODE REVIEW

File: tests/sparse_weights_test.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        with torch.no_grad():
+            for percent_on in [0.1, 0.5, 0.9]:
+                linear = torch.nn.Linear(in_features=in_features,
+                                         out_features=out_features)
+                sparse = SparseWeights(linear, percent_on)
+                nonzeros = torch.nonzero(sparse.module.weight, as_tuple=True)[0]
+                counts = torch.unique(nonzeros, return_counts=True)[1]
+
+                # Expected activity per unit
+                expected = [round(in_features * percent_on)] * out_features
```
Comment (subutai) [inline tests/sparse_weights_test.py]: This should check that each out_feature has `round(in_features * percent_on)` non-zero weights.

File: tests/sparse_weights_test.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            for percent_on in [0.1, 0.5, 0.9]:
+                cnn = torch.nn.Conv2d(in_channels=in_channels,
+                                      out_channels=out_channels,
+                                      kernel_size=kernel_size)
+                sparse = SparseWeights2d(cnn, percent_on)
+                nonzeros = torch.nonzero(sparse.module.weight, as_tuple=True)[0]
+                counts = torch.unique(nonzeros, return_counts=True)[1]
+
+                # Expected activity per channel
+                expected = [round(input_size * percent_on)] * out_channels
```
Comment (subutai) [inline tests/sparse_weights_test.py]: This should check that each out_channel has round(input_size * percent_on) non-zero weights.

File: tests/sparse_weights_test.py (hunk 1 above)
Comment (lscheinkman) [reply tests/sparse_weights_test.py]: That is exactly what it is being checked.
The `counts` tensor contains the count of nonzeros for each `out_features`. 
In the case where `in_features = 784` and `out_features=10` the tensor `counts` will contain the following:
```
[78, 78, 78, 78, 78, 78, 78, 78, 78, 78]
```

File: tests/sparse_weights_test.py (hunk 2 above)
Comment (lscheinkman) [reply tests/sparse_weights_test.py]: Same as before, the `counts` tensor contains the count of nonzeros for each output channel

File: tests/sparse_weights_test.py (hunk 1 above)
Comment (subutai) [reply tests/sparse_weights_test.py]: Ah ok, thanks.

**Pull request merged**